### PR TITLE
Sum of XOR Functions editorial

### PR DESCRIPTION
**Pull request by eysbutno**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Review by brebenelmihnea — DISMISSED** [review]

File: solutions/silver/cf-1879D.mdx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+---
+
+<Warning title="Be careful with integer overflow!">
+
+Because we are working with very large numbers, the problem statement asks us to
+take the answer by some very large modulo. For more details on how to handle modular
+arithmetic, you can read the [Modular Arithmetic](https://usaco.guide/gold/modular)
+on this guide.
+
+</Warning>
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1879D.mdx:15]: cool warning! this is usually taken for granted though, not sure if we should have this haha

File: solutions/silver/cf-1879D.mdx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+[Official Analysis (C++)](https://codeforces.com/blog/entry/120773)
+
+## Explanation
+Let's try to simplify the problem. Consider processing the array bit by bit.
+Each bit $x$ contributes $2^x \cdot \sum_{l=1}^{n} \sum_{r=l}^{n} p \cdot
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1879D.mdx:22]: have you considered adding hints?

File: solutions/silver/cf-1879D.mdx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+const int MOD = 998244353;
+int main() {
+	int n;
+	cin >> n;
+	vector<int> a(n + 1);
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1879D.mdx:50]: rename to `arr`

Comment (SansPapyrus683) [reply solutions/silver/cf-1879D.mdx:50]: also why's it 1-indexed

File: solutions/silver/cf-1879D.mdx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	for (int i = 1; i <= n; i++) { cin >> a[i]; }
+
+	ll res = 0;
+	for (int bit = 0; bit < 32; bit++) {
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1879D.mdx:54]:
```suggestion
	for (int bit = 0; bit < sizeof(int) * 8; bit++) {
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
massive

File: solutions/silver/cf-1879D.mdx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+Let's try to simplify the problem. Consider processing the array bit by bit.
+Each bit $x$ contributes $2^x \cdot \sum_{l=1}^{n} \sum_{r=l}^{n} p \cdot
+(r - l + 1)$ to the answer, where $p$ is the parity of this bit's occurrences.
+Because of this, it becomes apparent that we need to find the sum of subarray
+lengths for all subarrays that have this bit show up an odd amount of times.
+
+For an index $i$, we want to find the sum of subarray lengths for all relevant
+subarrays that end with $i$. This sum can be expressed as
+$ \sum_{l=1}^{x} i-v[l]+1$, where $v$ is the set of all valid left
+endpoints, and $x$ is the number of such endpoints. This expression is easily
```
Comment (envyaims) [inline solutions/silver/cf-1879D.mdx:30]: stop using easily pls
we are a wholesome organization

File: solutions/silver/cf-1879D.mdx (hunk 3 above)
Comment (envyaims) [reply solutions/silver/cf-1879D.mdx:50]: cope bro

File: solutions/silver/cf-1879D.mdx (hunk 4 above)
Comment (envyaims) [reply solutions/silver/cf-1879D.mdx:54]: wtf

**Review by envyaims — CHANGES_REQUESTED** [review]

File: solutions/silver/cf-1879D.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-1879D.mdx:30]: ^^^ thanks for catching that

File: solutions/silver/cf-1879D.mdx (hunk 3 above)
Comment (TheGamingMousse) [reply solutions/silver/cf-1879D.mdx:50]: > also why's it 1-indexed

1 indexed makes some of the math a bit easier

File: solutions/silver/cf-1879D.mdx (hunk 1 above)
Comment (TheGamingMousse) [reply solutions/silver/cf-1879D.mdx:15]: ok i'll remove it

File: solutions/silver/cf-1879D.mdx (hunk 2 above)
Comment (TheGamingMousse) [reply solutions/silver/cf-1879D.mdx:22]: hmmmm i might do that

File: solutions/silver/cf-1879D.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-1879D.mdx:54]: envy when he sees anything he doesn't understand

File: solutions/silver/cf-1879D.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-1879D.mdx:50]: perhaps make note of that then

File: solutions/silver/cf-1879D.mdx (hunk 4 above)
Comment (TheGamingMousse) [reply solutions/silver/cf-1879D.mdx:54]: tbh that expression would probably be more confusing than just having 32

Comment (SansPapyrus683) [reply solutions/silver/cf-1879D.mdx:54]: how so? sizeof just gives the # of bytes, and 8 is the # of bits in a byte

Comment (TheGamingMousse) [reply solutions/silver/cf-1879D.mdx:54]: true, it's not that complicated. i'll change it

**Review by envyaims — DISMISSED** [review]
AC

**Comment by TheGamingMousse** [comment]
> AC

https://codeforces.com/contest/1879/submission/258621032

File: solutions/silver/cf-1879D.mdx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+</Spoiler>
+
+<Spoiler title = "Explanation">
+
+## Explanation
+
+Let's try to simplify the problem. Consider processing the array bit by bit.
+The $x$'th least significant bit contributes $2^x \cdot \sum_{l=1}^{n} \sum_{r=l}^{n} p \cdot
+(r - l + 1)$ to the answer, where $p$ is the parity of this bit's occurrences.
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1879D.mdx:31]: i mean this is redundant now given this hints

File: content/3_Silver/Intro_Bitwise.problems.json [hunk 7]
```diff
@@ -120,8 +120,7 @@
       "isStarred": false,
       "tags": ["Bitwise", "Prefix Sums"],
       "solutionMetadata": {
-        "kind": "autogen-label-from-site",
-        "site": "CF"
+        "kind": "internal"
```
Comment (SansPapyrus683) [inline content/3_Silver/Intro_Bitwise.problems.json:123]: i think there's a metadata for like `hasHints` or something

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/silver/cf-1879D.mdx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+		parity_sum[1] = {0, 0};
+		int parity = 0;
+
+		for (int i = 1; i <= n; i++) {
+			if ((a[i] >> bit) & 1) { parity = (parity + 1) & 1; }
+			auto [dist, occ] = parity_sum[!parity];
+			len_sum = (len_sum + occ * i - dist) % MOD;
+			if (len_sum < 0) { len_sum += MOD; }
+			parity_sum[parity][0] += i;
+			parity_sum[parity][1]++;
```
Comment (ryanchou-dev) [inline solutions/silver/cf-1879D.mdx:78]: some comments here would be nice

**Review by ryanchou-dev — DISMISSED** [review]

File: solutions/silver/cf-1879D.mdx (hunk 8 above)
Comment (TheGamingMousse) [reply solutions/silver/cf-1879D.mdx:78]: ok added those

File: solutions/silver/cf-1879D.mdx (hunk 6 above)
Comment (TheGamingMousse) [reply solutions/silver/cf-1879D.mdx]: removed that first part

**Review by SansPapyrus683 — APPROVED** [review]

**Review by brebenelmihnea — APPROVED** [review]

**Pull request merged**